hoomd/mpcd/RotatedParallelPlateGeometry.h

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2009-2024 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+/*!
+ * \file mpcd/RotatedParallelPlateGeometry.h
+ * \brief Definition of the MPCD rotated slit channel geometry
+ */
+
+#ifndef MPCD_ROTATED_PARALLEL_PLATE_GEOMETRY_H_
+#define MPCD_ROTATED_PARALLEL_PLATE_GEOMETRY_H_
+
+#include "hoomd/BoxDim.h"
+#include "hoomd/HOOMDMath.h"
+
+#ifdef __HIPCC__
+#define HOSTDEVICE __host__ __device__ inline
+#else
+#define HOSTDEVICE inline __attribute__((always_inline))
+#include <string>
+#endif // __HIPCC__
+
+namespace hoomd
+    {
+namespace mpcd
+    {
+//! Rotated parallel plate (slit) geometry
+/*!
+ */
+class __attribute__((visibility("default"))) RotatedParallelPlateGeometry
+    {
+    public:
+    //! Constructor
+    /*!
+     * \param H Channel half-width
+     * \param angle Rotation angle (degrees)
+     * \param V Velocity of the wall
+     * \param no_slip Boundary condition at the wall (slip or no-slip)
+     */
+    HOSTDEVICE RotatedParallelPlateGeometry(Scalar H, Scalar angle, Scalar V, bool no_slip)
+        : m_H(H), m_angle(angle * M_PI / 180.), m_V(V), m_no_slip(no_slip)
+        {
+        }
+
+    //! Detect collision between the particle and the boundary
+    /*!
+     * \param pos Proposed particle position
+     * \param vel Proposed particle velocity
+     * \param dt Integration time remaining
+     *
+     * \returns True if a collision occurred, and false otherwise
+     *
+     * \post The particle position \a pos is moved to the point of reflection, the velocity \a vel
+     * is updated according to the appropriate bounce back rule, and the integration time \a dt is
+     * decreased to the amount of time remaining.
+     */
+    HOSTDEVICE bool detectCollision(Scalar3& pos, Scalar3& vel, Scalar& dt) const
+        {
+        // TODO: fill this in
+        const Scalar y1 = pos.x * tan(m_angle * M_PI / 180) + m_H;
+        const Scalar y2 = pos.x * tan(m_angle * M_PI / 180) - m_H;
+
+        const signed char sign = (char)((pos.y > y1) - (pos.y < y2));
+
+        if (sign == 0 || (vel.x * tan(m_angle * M_PI) - vel.y) == Scalar(0))
+            {
+            dt = Scalar(0);
+            return false;
+            }
+
+        if (sign == +1)
+            {
+            dt = (pos.y - y1) / (vel.x * tan(m_angle * M_PI) - vel.y);
+
+            pos.x -= vel.x * dt;
+            pos.y = y1;
+            pos.z -= vel.z * dt;
+            }
+
+        if (sign == -1)
+            {
+            dt = (y2 - pos.y) / (vel.x * tan(m_angle * M_PI) - vel.y);
+
+            pos.x -= vel.x * dt;
+            pos.y = y2;
+            pos.z -= vel.z * dt;
+            }
```

Comment:
I think this would be equivalent to these two if branches? Basically, select which wall based on the sign, then also use the sign in the calculation.
```suggestion
        const Scalar y_wall = (sign == 1) ? y1 : y2;
        dt = sign * (pos.y - y_wall) / v_normal;

        pos.x -= vel.x * dt;
        pos.y = y_wall;
        pos.z -= vel.z * dt;
```